sway-fmt-v2/src/utils/comments.rs

```diff
@@ -69,6 +100,308 @@ fn collect_comments_from_token_stream(
+        &formatted_module,
+        formatted_code,
+        unformatted_input,
+    )?;
+    Ok(())
+}
+/// Adds the comments from comment_map to correct places in the formatted code. This requires us
+/// both the unformatted and formatted code's items as they will have different spans for their
+/// nodes. While traversing the unformatted items, `add_comments` searches for comments. If there is a comment found
+/// places the comment.
+fn add_comments(
+    comment_map: CommentMap,
+    unformatted_module: &Module,
+    formatted_module: &Module,
+    formatted_code: &mut FormattedCode,
+    unformatted_code: Arc<str>,
+) -> Result<(), FormatterError> {
+    let unformatted_comment_spans = unformatted_module.collect_spans();
+    let formatted_comment_spans = formatted_module.collect_spans();
+
+    // Since we are adding comments into formatted code, in the next iteration the spans we find for the formatted code needs to be offsetted
+    // as the total length of comments we added in previous iterations.
+    let mut offset = 0;
+
+    // We will definetly have a span in the collected span since for a source code to be parsed there should be some tokens present.
+    let mut previous_unformatted_comment_span = unformatted_comment_spans
+        .first()
+        .ok_or(FormatterError::CommentError)?;
+    let mut previous_formatted_comment_span = formatted_comment_spans
+        .first()
+        .ok_or(FormatterError::CommentError)?;
+    for (unformatted_comment_span, formatted_comment_span) in unformatted_comment_spans
+        .iter()
+        .skip(1)
+        .zip(formatted_comment_spans.iter().skip(1))
+    {
+        let comments_found = get_comments_between_spans(
+            previous_unformatted_comment_span,
+            unformatted_comment_span,
+            &comment_map,
+            &unformatted_code,
+        );
+        if !comments_found.is_empty() {
+            offset += insert_after_span(
+                previous_formatted_comment_span,
+                comments_found,
+                offset,
+                formatted_code,
+            )?;
+        }
+        previous_unformatted_comment_span = unformatted_comment_span;
+        previous_formatted_comment_span = formatted_comment_span;
+    }
+    Ok(())
+}
+
+// A CommentWithContext is the Comment and the offset before it. The offset can be between the (from) item we searched for this comment or from the last comment inside range
+type CommentWithContext = (Comment, String);
+
+/// Returns a list of comments between given spans. For each comment returns the Context
+/// Context of a comment is basically the offset (the characters between the last item/comment) to the current comment
+fn get_comments_between_spans(
+    from: &CommentSpan,
+    to: &CommentSpan,
+    comment_map: &CommentMap,
+    unformatted_code: &Arc<str>,
+) -> Vec<CommentWithContext> {
+    let mut comments_with_context = Vec::new();
+    for (index, comment_tuple) in comment_map
```

Comment:
We should never need to write `comment_tuple`, as `tuple` and tuple indexing (e.g. `.0`, `.1`, etc) is unnecessarily ambiguous for the reader in the case they don't happen to know what the tuple contains. Instead, we could match this pattern further, e.g.

```rust
for (index, (comment_span, comment)) in comment_map
```

I think it's good to keep in mind that while useful in some niche cases, tuple indexing is generally a bit of a code smell, particularly when a more descriptive binding name could be given to the elements we care about. We should always try to prioritize the reader of the code ahead of the writer.